src/game/Chain.java

```diff
@@ -0,0 +1,82 @@
+package game;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * A chain consists of stones (intersections with the same Mark, B or W) Each
+ * stone in the chain has the same number of liberties, namely the amount of
+ * neighbouring intersections of the chain that have Mark U.
+ */
+public class Chain {
```

Comment:
I think the `Chain` concept is already one of the stronger parts in your application. It could maybe have been even cooler if it was setup as a recursive pattern: Than the chains would not be have to be something "static" you join. Every intersection on the board would have `inSameChain` method, that calls the `inSameChain` method on all it's neighboring stones. 

Or something along those lines :-)